File: frontend/src/objects/order/core/OrderSummaryReviewState.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  IsoDateTime,
  RatingValue,
  ReasonText,
  ReviewStatus,
} from '@/objects/core/SharedObjects'

export type OrderSummaryReviewState = {
  storeRating?: RatingValue
  riderRating?: RatingValue
  merchantRejectReason?: ReasonText
  reviewStatus: ReviewStatus
  reviewRevokedReason?: ReasonText
  reviewRevokedAt?: IsoDateTime
}

File: backend/src/main/scala/domain/order/OrderSummaryReviewState.scala
package domain.order

import domain.shared.*

final case class OrderSummaryReviewState(
    storeRating: Option[RatingValue],
    riderRating: Option[RatingValue],
    merchantRejectReason: Option[ReasonText],
    reviewStatus: ReviewStatus,
    reviewRevokedReason: Option[ReasonText],
    reviewRevokedAt: Option[IsoDateTime],
)
